Utils/Editor/UpdateListTests.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using DUCK.Utils;
+using NUnit.Framework;
+
+namespace DUCK.Utils.Editor
+{
+	[TestFixture]
+	public class UpdateListTest
+	{
+		[Test]
+		public void ExpectConstructorNotToThrow()
+		{
+			Assert.DoesNotThrow(() =>
+			{
+				new UpdateList();
+			});
+		}
+
+		[Test]
+		public void ExpectAddedFunctionToBeCalledWhenUpdated()
+		{
+			var wasCalled = false;
+			Action<float> func = dt =>
+			{
+				wasCalled = true;
+			};
+
+			var updateList = new UpdateList();
+			updateList.Add(func);
+			Assert.IsFalse(wasCalled);
+			updateList.Update(0f);
+			Assert.IsTrue(wasCalled);
+		}
+
+		[Test]
+		public void ExpectAddToThrowIfFunctionIsAlreadyAdded()
+		{
+			Action<float> func = dt => {};
+			var updateList = new UpdateList();
+			updateList.Add(func);
+			Assert.Throws<Exception>(() =>
+			{
+				updateList.Add(func);
+			});
+		}
+
+		[Test]
+		public void ExpectContainsToReturnTrueOnlyIfFunctionWasAdded()
+		{
+			Action<float> func = dt => {};
+			var updateList = new UpdateList();
+			Assert.IsFalse(updateList.Contains(func));
+			updateList.Add(func);
+			Assert.IsTrue(updateList.Contains(func));
+		}
+
+		[Test]
+		public void ExpectRemoveToThrowIfFunctionWasNotFound()
+		{
+			Action<float> func = dt => {};
+			var updateList = new UpdateList();
+
+			Assert.Throws<Exception>(() =>
+			{
+				updateList.Remove(func);
+			});
+		}
+
+		[Test]
+		public void ExpectFunctionToNotBeCalledIfRemoveWasCalled()
+		{
+			var wasCalled = false;
+			Action<float> func = dt =>
+			{
+				wasCalled = true;
+			};
+
+			var updateList = new UpdateList();
+			updateList.Add(func);
+			Assert.IsFalse(wasCalled);
+			updateList.Remove(func);
+			Assert.IsFalse(wasCalled);
+
+			updateList.Update(0f);
+			Assert.IsFalse(wasCalled);
+		}
+
+		[Test]
+		public void ExpectContainsToReturnFalseAfterRemoveWasCalled()
+		{
+			Action<float> func = dt => { };
+			var updateList = new UpdateList();
+			updateList.Add(func);
+			updateList.Remove(func);
+			Assert.IsFalse(updateList.Contains(func));
+		}
+
+		[Test]
+		public void ExpectContainsToReturnFalseAfterRemoveWasCalledDuringFrame()
+		{
+			// we have to create another ref to the function so we can self remove within closure
+			Action<float> funcRef = null;
+
+			var updateList = new UpdateList();
+
+			Action<float> func = dt =>
+			{
+				updateList.Remove(funcRef);
+
+				Assert.IsFalse(updateList.Contains(funcRef));
+			};
+
+			funcRef = func;
+
+			updateList.Add(funcRef);
```

Comment:
Is this missing the `updateList.Update(0f);` ?

Reply:
Yes!